### PR TITLE
Add codecov.io

### PR DESCRIPTION
**Pull request by gliptak**
You will likely have to login and enable the project at https://codecov.io/

Here is the report generated for my branch:

https://codecov.io/gh/gliptak/gb/branch/codecov1

### CODE REVIEW

File: bin/test [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+
+go get -t ./...
+
+rm -f coverage.txt
```
Comment (davecheney) [inline bin/test]: this shouldn't be needed

File: bin/test [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+for d in $(find . -not \( -name "*testdata" -prune \) -name '*.go' -exec dirname {} \; | sort -u); do
```
Comment (davecheney) [inline bin/test]: go list github.com/constabulary/gb/...

File: .travis.yml [hunk 3]
```diff
@@ -19,10 +19,14 @@ addons:
...
   - go get -t -v ./...
   - git clone --quiet --single-branch --depth 1 https://github.com/constabulary/integration-tests.git ../integration-tests
+  - if [[ $TRAVIS_GO_VERSION == 1.4* ]]; then go get -u golang.org/x/tools/cmd/cover; fi
   - echo '#!/bin/bash' > "$GOPATH/bin/sudo"
     && echo 'echo >&2 attempted sudo "$@"' >> "$GOPATH/bin/sudo"
     && chmod +x "$GOPATH/bin/sudo"
 
 script:
-  - go test -v ./...
+  - bin/test
```
Comment (davecheney) [inline .travis.yml]: please call this "bin/coverage.sh" or something more meaningful

File: bin/test [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+
+go get -t ./...
```
Comment (davecheney) [inline bin/test]: not needed

**Comment by gliptak** [comment]
Would you like see additional changes? Thanks

**Comment by davecheney** [comment]
I probably won't have time to review this til the weekend.

**Comment by davecheney** [comment]
I had a quick look at this and it looks like there are some files missing
```
0.16s$ bash <(curl -s https://codecov.io/bash)
  _____          _
 / ____|        | |
| |     ___   __| | ___  ___ _____   __
| |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
| |___| (_) | (_| |  __/ (_| (_) \ V /
 \_____\___/ \__,_|\___|\___\___/ \_/
                              Bash-31d77b9
==> Travis CI detected.
    project root: .
    Yaml not found, that's ok! Learn more at http://docs.codecov.io/docs/codecov-yaml
==> Running gcov in . (disable via -X gcov)
==> Searching for coverage reports in:
    + .
==> Python coveragepy not found
--> No coverage report found.
    Please visit http://docs.codecov.io/docs/supported-languages
```

**Comment by gliptak** [comment]
@davecheney I'm not sure what to make of those generic messages above. If you look at the bottom of

https://codecov.io/gh/constabulary/gb/pull/664?src=pr

you will see listing of the covered files. I can submit a folloup PR to fine-tune.

**Comment by davecheney** [comment]
I'm not convinced this is working, but there's no point in arguing about it in a PR.

**Comment by davecheney** [comment]
@gliptak can you please take a look, I don't think this is working.

**Comment by gliptak** [comment]
@davecheney You are correct, it is not working.

My PR replaced ```go test``` with ``coverage.sh```:

https://github.com/constabulary/gb/pull/664/commits/01140f64846f2ad184e12f8a2bb4ba33d0a52af1#diff-354f30a63fb0907d4ad57269548329e3R28

The version committed to master, no longer does:

https://github.com/constabulary/gb/pull/664/commits/a364c1a64b908cb6bffbe2a3193ce321131d7717#diff-354f30a63fb0907d4ad57269548329e3R29

**Comment by davecheney** [comment]
Sorry, that was my fault when I tried to resolve the merge conflict yesterday.

Please send a new PR.

**Pull request merged**